MQTicketService/Dockerfile

```diff
@@ -11,37 +11,48 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM openjdk:8-alpine
+FROM adoptopenjdk/openjdk11:alpine
 USER root
 RUN apk update && \
     apk add wget;
 #
 COPY ./TicketGenerator /data/TicketGenerator
 #
 # Setup JAVA_HOME -- useful for docker commandline
-ENV JAVA_HOME /usr/lib/jvm/java-8-openjdk-amd64/
+ENV JAVA_HOME /usr/lib/jvm/java-11-openjdk-amd64/
 RUN export JAVA_HOME
 #
 # Compile the source code
 RUN cd /data/TicketGenerator && \
     mkdir jarFiles && \
     cd jarFiles && \
-    wget https://repo1.maven.org/maven2/com/ibm/mq/com.ibm.mq.allclient/9.2.0.0/com.ibm.mq.allclient-9.2.0.0.jar && \
+    wget https://repo1.maven.org/maven2/com/ibm/mq/com.ibm.mq.allclient/9.2.4.0/com.ibm.mq.allclient-9.2.4.0.jar && \
     wget https://repo1.maven.org/maven2/javax/jms/javax.jms-api/2.0.1/javax.jms-api-2.0.1.jar && \
     wget https://repo1.maven.org/maven2/org/json/json/20190722/json-20190722.jar && \
+    wget https://repo1.maven.org/maven2/jakarta/xml/bind/jakarta.xml.bind-api/3.0.0/jakarta.xml.bind-api-3.0.0.jar && \
+    wget https://repo1.maven.org/maven2/org/glassfish/jaxb/jaxb-core/3.0.0/jaxb-core-3.0.0.jar && \
+    wget https://repo1.maven.org/maven2/com/sun/xml/bind/jaxb-impl/3.0.0/jaxb-impl-3.0.0.jar && \
+    wget https://repo1.maven.org/maven2/com/sun/istack/istack-commons-runtime/4.0.1/istack-commons-runtime-4.0.1.jar && \
+    wget https://repo1.maven.org/maven2/jakarta/activation/jakarta.activation-api/2.0.1/jakarta.activation-api-2.0.1.jar && \
     cd /data/TicketGenerator && \
-    javac -cp /data/TicketGenerator/jarFiles/*: /data/TicketGenerator/com/ibm/mq/badge/*.java;
+    javac -cp /data/TicketGenerator/jarFiles/*: com/ibm/mq/badge/*.java;
 #
 #
 # Switch to the MQ image
 # Tested with
+#     9.2.4.0-r1
```

Comment:
Changed as required.

Reply:
There is no `mvn package` which suggests that the changes have not been tested.

Reply:
Added `mvn clean package` as required in the Dockerfile. Also added a `.dockerignore` so that the Dockerfile does not copy the TicketGenerator uber jar created locally.